arcs/src/systems/spatial_relation.rs

```diff
@@ -0,0 +1,204 @@
+use crate::{
+    components::{SpatialEntity, DrawingObject, Space},
+    algorithms::{Bounded},
+};
+use specs::prelude::*;
+
+/// A [`System`] which keeps track of the spatial relation of entities
+#[derive(Debug)]
+pub struct SpatialRelation {
+    changes: ReaderId<ComponentEvent>,
+    to_insert: BitSet,
+    to_update: BitSet,
+}
+
+impl SpatialRelation {
+    pub const NAME: &'static str = module_path!();
+
+    pub fn new(world: &World) -> Self {        
+        SpatialRelation {
+            changes: world.write_storage::<DrawingObject>().register_reader(),
+            to_insert: BitSet::new(),
+            to_update: BitSet::new(),
+        }
+    }
+}
+
+impl<'world> System<'world> for SpatialRelation {
+    type SystemData = (
+        Write<'world, Space>,
+        ReadStorage<'world, DrawingObject>,
+        Entities<'world>
+    );
+
+    fn run(&mut self, data: Self::SystemData) {
+        // clear any left-over flags
+        self.to_insert.clear();
+        self.to_update.clear();
+
+        let (mut space, drawing_objects, entities) = data;
+
+        // find out which items have changed since we were last polled
+        for event in drawing_objects.channel().read(&mut self.changes) {
+            println!("Event is {:#?}", event);
```

Comment:
I'm guessing we'll want to remove this before merging.

Reply:
Oh yes, I will search the code for all `println!`